### ISSUE TITLE
External subtitles out of sync when crossing DVD chapter boundaries

### ISSUE THREAD

**Issue by mvglasow** [xbmc/xbmc#23430]
### Describe the bug
Here is a clear and concise description of what the problem is:

When playing a DVD with external subtitles (e.g. from a .srt file), subtitles will go out of sync at the next chapter transition and restart from the beginning.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The main DVD track plays, and correct subtitles are shown throughout its entire length.

## Actual Behavior

The first chapter plays with the correct subtitles. At the end of the first chapter, the next chapter will play, but the subtitles will restart at the first chapter. (If you use a ”timestamp” SRT file as described below, it will jump back to 00:00:00 at each new chapter.)

What external subtitles are (re-)added in the middle of the film, the currently playing chapter will have the correct subtitles (regardless of whether the currently playing chapter is the first or a later one), but the sync issue described above will occur at the next chapter transition.

## Possible Fix

As far as I know, .srt has no support for chapters and timestamps are relative to the beginning of the entire video stream, not the current chapter. This seems to get honored when the .srt file is loaded, regardless of which chapter is currently playing. However, on chapter transition, the timestamp for subtitles is incorrectly reset to 00:00:00.

### To Reproduce

1. Copy the entire directory tree from a DVD into a new directory on one of Kodi’s movie library paths. (In my case this was on an SMB share, not sure if that’s relevant. Might also be reproducible when playing directly from DVD.)
2. Place an external .srt file in that same directory. (Not sure if the path is relevant.)
3. Start playing the main DVD track, and load subtitles from the external .srt file.
4. Watch beyond at least one chapter change, and follow subtitles.

Timestamps in the external .srt file must be relative to start of track (not chapter). For a quick test, you can build a .srt file like the following:

```
1
00:00:00,000 --> 00:00:00,999
00:00:00

2
00:00:01,000 --> 00:00:01,999
00:00:01
```
etc. for the entire length of the movie – this will give you timestamps as subtitles.

### Debuglog

The debuglog can be found here:
Currently none. The issue should be reproducible; let me know if you cannot reproduce the issue and I will provide a log.

### Screenshots 
Here are some links or screenshots to help explain the problem:

Currently none.

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

n/a

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: OSMC 2023.01-1
 - Kodi version: 19.5.0

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by mvglasow** [comment]
The issue seems to be fixed as of 20.1.0, OSMC 2023.06-1.

**Comment by enen92** [comment]
Thanks for confirming.
